lightning/src/ln/channel.rs

```diff
@@ -3352,44 +3376,125 @@ impl<Signer: Sign> Channel<Signer> {
 		}
 	}
 
-	fn maybe_propose_first_closing_signed<F: Deref>(&mut self, fee_estimator: &F) -> Option<msgs::ClosingSigned>
+	/// Calculates and returns our minimum and maximum closing transaction fee amounts, in whole
+	/// satoshis. The amounts remain consistent unless a peer disconnects/reconnects or we restart,
+	/// at which point they will be recalculated.
+	fn calculate_closing_fee_limits<F: Deref>(&mut self, fee_estimator: &F) -> (u64, u64)
 		where F::Target: FeeEstimator
 	{
-		if !self.is_outbound() || !self.pending_inbound_htlcs.is_empty() || !self.pending_outbound_htlcs.is_empty() ||
-				self.channel_state & (BOTH_SIDES_SHUTDOWN_MASK | ChannelState::AwaitingRemoteRevoke as u32) != BOTH_SIDES_SHUTDOWN_MASK ||
-				self.last_sent_closing_fee.is_some() || self.pending_update_fee.is_some() {
-			return None;
-		}
+		if let Some((min, max)) = self.closing_fee_limits { return (min, max); }
 
+		// Propose a range from our current Background feerate to our Normal feerate plus our
+		// force_close_avoidance_max_fee_satoshis.
+		// If we fail to come to consensus, we'll have to force-close.
 		let mut proposed_feerate = fee_estimator.get_est_sat_per_1000_weight(ConfirmationTarget::Background);
```

Comment:
One hardening we could introduce in the future would be to hardcode an upper bound in case of compromise of our fee-estimator. The closing transaction max size is known and in the worst-case scenario the confirmation of a closing transaction is not a matter of safety.

Reply:
Hmm, yea, we've talked a few times about bounding the fee estimator results to "reasonable" values. We should probably do it sometime, but for now I'd say its fine.